GraphBulkImporter/Extensions.cs

```diff
@@ -0,0 +1,103 @@
+        /// </summary>
+        /// 
+        /// <returns>
+        /// A new GremlinVertex object 
+        /// with its id and partitionKey properties set to the corresspinding id and partitionKey properties of the object being converted
+        /// and the label property set to the type name of the object
+        /// </returns>
+        public static GremlinVertex ToGremlinVertex(this object obj)
+        {
+            var label = obj.GetType().Name;
+            
+            if (!obj.HasProperty("id"))
+            {
+                throw new ArgumentException($"{label} does not have expected property.", "id");
+            }
+
+            if (!obj.HasProperty("partitionKey"))
+            {
+                throw new ArgumentException($"{label} does not have expected property.", "partitionKey");
+            }
+
+            return ToGremlinVertex(obj, "id", "partitionKey", label);
+        }
+
+        /// <summary>
+        /// Converts an object to a GremlinVertex allowing more control over how the id, partitionKey, and label properties are set. 
+        /// </summary>
+        /// <param name="idProperty">Which property of the object should be used for id of the GremlinVertex, defaults to "id"</param>
+        /// <param name="partitionKeyProperty">Which property should be used for partitionKey of the GremlinVertex, defaults to "partitionKey"</param>
+        /// <param name="label">What value should be used for the label property, defaults to the Type name of the object</param>
+        /// <returns>
+        /// A new instance of a GremlinVertex
+        /// with its id and partitionKey property values set to and <typeparamref name="idProperty"/> and <typeparamref name="partitionKeyProeprty"/> properties respectively
+        /// and the label property set to the value of <typeparamref name="label"/>
+        /// </returns>
+        public static GremlinVertex ToGremlinVertex(this object obj, string idProperty,  string partitionKeyProperty, string label)
+        {
+            if (string.IsNullOrEmpty(idProperty))
+            {
+                throw new ArgumentException($"{nameof(idProperty)} cannot be Null or Empty", nameof(idProperty));
+            }
+
+            if (string.IsNullOrWhiteSpace(partitionKeyProperty))
+            {
+                throw new ArgumentException($"{nameof(partitionKeyProperty)} cannot be Null or Empty", nameof(partitionKeyProperty));
+            }
+
+            if (string.IsNullOrWhiteSpace(label))
+            {
+                throw new ArgumentException($"{nameof(label)} cannot be Null or Empty", nameof(label));
+            }
+
+            var gv = new GremlinVertex(obj.GetPropertyValue(idProperty).ToString(), label);
+            gv.AddProperty(new GremlinVertexProperty("partitionKey", obj.GetPropertyValue(partitionKeyProperty)));
+
+            foreach (var prop in obj.GetType().GetProperties(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance))
```

Comment:
BindingFlags.NonPublic or BindingFlags.Static? do you want to copy out the no public fields like protected, internal and private?

Reply:
removed binding flags entirely here, so should get all Properties. 
the other occasion where i use BindingFlags is to set IgnoreCase, but that's not the case here.